File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(const string& s) {
        
        if (s.size() <= 1) { return s.size(); }
        
        int chars[256] { 0 }; 
        
        int left = 0, right = 0, maxSize = INT_MIN, currentSize = 0;
        
        while (right != s.size()) {
            
            ++chars[s[right]];
            
            ++currentSize;
            
            if (chars[s[right]] > 1) {
                
                while (chars[s[right]] > 1) {

                    --chars[s[left]];

                    ++left;
                    
                    --currentSize;
                }
            }
            
            maxSize = currentSize > maxSize ? currentSize : maxSize;
            
            ++right;
        }
        
        return maxSize;
    }
};

File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        if (s.length() <= 1) { return s.length(); }
        
        int chars[] = new int[256]; 
        
        int left = 0, right = 0, maxSize = Integer.MIN_VALUE, currentSize = 0;
        
        while (right != s.length()) {
            
            ++chars[s.charAt(right)];
            
            ++currentSize;
            
            if (chars[s.charAt(right)] > 1) {
                
                while (chars[s.charAt(right)] > 1) {

                    --chars[s.charAt(left)];

                    ++left;
                    
                    --currentSize;
                }
            }
            
            maxSize = currentSize > maxSize ? currentSize : maxSize;
            
            ++right;
        }
        
        return maxSize;
    }
}